resources/src/operation/mod.rs

```diff
@@ -252,95 +255,112 @@ impl<'de> Deserialize<'de> for Operation {
-                }
+                return Err(de::Error::custom(
+                    "Missing fields for create account operation.",
+                ))
             }
+
             IntermediateOperation {
-                operation_type,
+                operation_type: "path_payment",
                 from: Some(from),
                 to: Some(to),
+                asset_code,
+                asset_issuer,
                 asset_type: Some(asset_type),
                 amount: Some(amount),
+                source_asset_code,
+                source_asset_issuer,
                 source_asset_type: Some(source_asset_type),
                 source_amount: Some(source_amount),
                 source_max: Some(source_max),
                 ..
             } => {
-                if operation_type == "path_payment" {
-                    let destination_asset_identifier = AssetIdentifier::new(
-                        &asset_type,
-                        rep.asset_code,
-                        rep.asset_issuer,
-                    ).map_err(|_| {
-                        de::Error::custom("Code and issuer are required for non-native assets")
-                    })?;
-                    let source_asset_identifier = AssetIdentifier::new(
-                        &source_asset_type,
-                        rep.source_asset_code,
-                        rep.source_asset_issuer,
-                    ).map_err(|_| {
-                        de::Error::custom("Code and issuer are required for non-native assets")
-                    })?;
-                    Some(OperationDetail::PathPayment(
-                        self::path_payment::PathPayment::new(
-                            from,
-                            to,
-                            destination_asset_identifier,
-                            amount,
-                            source_asset_identifier,
-                            source_amount,
-                            source_max,
-                        ),
-                    ))
-                } else {
-                    None
-                }
+                let destination_asset_identifier =
+                    AssetIdentifier::new(&asset_type, asset_code, asset_issuer)
+                        .map_err(|err| de::Error::custom(err))?;
+                let source_asset_identifier = AssetIdentifier::new(
+                    &source_asset_type,
+                    source_asset_code,
+                    source_asset_issuer,
+                ).map_err(|err| de::Error::custom(err))?;
+                OperationDetail::PathPayment(PathPayment::new(
+                    from,
+                    to,
+                    destination_asset_identifier,
+                    amount,
+                    source_asset_identifier,
+                    source_amount,
+                    source_max,
+                ))
+            }
+            IntermediateOperation {
+                operation_type: "path_payment",
```

Comment:
I'm debating moving all the error cases to the end of the match block. What do you think?